File: JAVA6_ASM/src/main/java/com/poly/service/DonHangService.java
package com.poly.service;

import java.util.List;

import com.fasterxml.jackson.databind.JsonNode;
import com.poly.entity.HoaDon;
import com.poly.entity.Hoadonchitiet;
import com.poly.entity.SanPham;

public interface DonHangService {
	public List<HoaDon> findAll();

	public HoaDon findById(Integer idHd);

	public HoaDon create(HoaDon donhang);
	
	public Hoadonchitiet createHDCT(Hoadonchitiet hdct);
	
	public HoaDon create1(JsonNode orderData);
	
	public HoaDon update(HoaDon donhang);

	public List<HoaDon> findAllByTaiKhoan(String taiKhoan);

	List<SanPham> getChiTietDonHang(Integer idHd);
	
}

File: JAVA6_ASM/src/main/resources/static/js/adminJS/DonHangService.js
app.service('DonHangService', function() {
  var donhang = {}; // Đối tượng để lưu trữ dữ liệu donhang

  // Phương thức để đặt dữ liệu donhang
  this.setDonHang = function(data) {
    donhang = data;
  };

  // Phương thức để lấy dữ liệu donhang
  this.getDonHang = function() {
    return donhang;
  };
});